### ISSUE TITLE
Template functions with annotations / ng-strict-di

### ISSUE THREAD

**Issue by karlhorky** [toddmotto/angular-component#11]
As per [my GitHub comment in this angular.js issue](https://github.com/angular/angular.js/issues/13645#issuecomment-172034441), I'm running into issues with template functions with annotating or `ng-strict-di`.

I know this issue may belong in the main angular.js repo, but I just wanted to ask, in case someone here had an idea how to get around this.

**Comment by toddmotto** [comment]
Not an immediate fix, will look into this, but doing this works using `$inject`: https://jsfiddle.net/toddmotto/vy4ayzb8/

**Comment by toddmotto** [comment]
Just pushed a fix for this, see https://github.com/toddmotto/angular-component/commit/28ff08e3e6f16bc2cb39c8369e73bcb918a79814

[Demo for annotated code](https://jsfiddle.net/toddmotto/rktzkjvh).

This allows for:

``` js
// Array annotations with minification
template: ['$element', '$attrs', function template(a, b) {
  console.log(a, b); // $element, $attrs
  return [
    '<div class="counter">',
      '<p>Counter component</p>',
      '<input type="text" ng-model="counter.count">',
      '<button type="button" ng-click="counter.decrement();">-</button>',
      '<button type="button" ng-click="counter.increment();">+</button>',
    '</div>'
  ].join('');
}]
```

``` js
// No annotations with automated minification (see source code `isArray` ternary + annotations)
template: function template(a, b) {
  console.log(a, b); // $element, $attrs
  return [
    '<div class="counter">',
      '<p>Counter component</p>',
      '<input type="text" ng-model="counter.count">',
      '<button type="button" ng-click="counter.decrement();">-</button>',
      '<button type="button" ng-click="counter.increment();">+</button>',
    '</div>'
  ].join('');
}
```

``` js
// Basic template String
template: [
  '<div class="counter">',
    '<p>Counter component</p>',
    '<input type="text" ng-model="counter.count">',
    '<button type="button" ng-click="counter.decrement();">-</button>',
    '<button type="button" ng-click="counter.increment();">+</button>',
  '</div>'
].join('')
```

**Comment by karlhorky** [comment]
Thanks @toddmotto! Works great. :+1:

**Comment by toddmotto** [comment]
@karlhorky Sweet! I've published `0.0.5` which includes the new automatic annotations, for NPM as well. Closing, over and out.

**Comment by toddmotto** [comment]
@karlhorky FYI, I've just published `0.0.6`, which aligns with further `component()` changes, small Object merge rather than `if` statements: https://github.com/toddmotto/angular-component/commit/1229b256524cb840023db96e1c086af956c33f57

**Comment by karlhorky** [comment]
:100: thanks, sounds good
